src/apiClient.ts

```diff
@@ -6,30 +6,55 @@ export interface ApiClientConfig {
   optionServiceUrl?: string
   analyticsServiceUrl?: string
   userNotificationServiceUrl?: string
-  tokenRefreshServiceUrl?: string
   debug?: boolean
 }
 
-export interface Tokens {
-  accessToken?: string
-  refreshToken?: string
-}
-
 export interface Handlers {
   onAccessTokenUpdate?: (token: string) => void
   onFailedTokenRefresh?: () => void
 }
 
 class ApiClient {
-  static instance: ApiClient
+  private static instance: ApiClient
 
-  configuration: ApiClientConfig = {}
-  tokens: Tokens = {}
-  handlers: Handlers = {}
-  version = "v1"
+  configuration: ApiClientConfig
+  accessToken: {
+    token: string | null
+    expires: Date | null
+  }
+  handlers: Handlers
```

Comment:
I could imagine not grouping these in a `handlers` variable. passing `onFailedTokenRefresh` directly would make the interface clearer to me. I had a "what are handlers" moment and had to look at the types to figure out what to do

really optional - feel free to go with this if you have good reasons to